routing/pathfind.go

```diff
@@ -534,82 +550,99 @@ func findPath(tx *bolt.Tx, graph *channeldb.ChannelGraph,
 
 		// If this vertex or edge has been black listed, then we'll skip
 		// exploring this edge.
-		if _, ok := ignoredNodes[v]; ok {
+		if _, ok := ignoredNodes[fromVertex]; ok {
 			return
 		}
 		if _, ok := ignoredEdges[edge.ChannelID]; ok {
 			return
 		}
 
+		nodeInfo := distance[toNode]
+
+		amountToSend := nodeInfo.amountToReceive
+
+		// Compute fee based on the amount that needs to be send to the
+		// next node.
+		var fee lnwire.MilliSatoshi
+		if fromVertex != sourceVertex {
+			fee = computeFee(amountToSend, edge)
+		} else {
+			// Source node has no precedessor to pay fee
+			// and fee should therefore also not be included in
+			// fee limit check and edge weight.
+			fee = 0
+		}
+
+		// Check if accumulated fees would exceed fee limit when
+		// this node would be added to the path.
+		totalFee := amountToSend + fee - amt
+		if totalFee > feeLimit {
+			return
+		}
+
 		// Compute the tentative distance to this new channel/edge which
-		// is the distance to our pivot node plus the weight of this
-		// edge.
-		tempDist := distance[pivot].dist + edgeWeight(amt, edge)
+		// is the distance from our toNode to the target node plus the 
+		// weight of this edge.
+		tempDist := nodeInfo.dist + edgeWeight(fee, edge.TimeLockDelta)
 
 		// If this new tentative distance is better than the current
 		// best known distance to this node, then we record the new
 		// better distance, and also populate our "next hop" map with
 		// this edge. We'll also shave off irrelevant edges by adding
 		// the sufficient capacity of an edge and clearing their
 		// min-htlc amount to our relaxation condition.
-		if tempDist < distance[v].dist && bandwidth >= amt &&
-			amt >= edge.MinHTLC && edge.TimeLockDelta != 0 {
+		if tempDist < distance[fromVertex].dist && bandwidth >= amountToSend &&
```

Comment:
It would be nice to separate the conditions here by turning them into early returns for better readability:
```go
// comment
if amountToSend > bandwidth {
    return
}

// comment
if amountToSend < edge.MinHTLC {
    return
}

// comment
if tempDist >= distance[fromVertex].dist {
    return
}

etc.

```

Reply:
Fixed. Only not sure about the reason for checking timelock == 0. Do you know when this may happen?